### PR TITLE
Add salt-api on Windows platform

### PR DESCRIPTION
**Pull request by rares-pop**
salt-api is broken on Windows platform
due to pickling issues on the salt.loader.netapi
Workaround that by creating a runner class
that can be pickled.

Signed-off-by: Rares POP <rares.pop@ni.com>

### What does this PR do?
It adds/fixes salt-api on the Windows platform, using rest_cherrypy module

### Previous Behavior
salt-api wasn't available on Windows

### New Behavior
The user is able to run the salt-api daemon on windows, using rest_cherrypy module

### Tests written?

No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by gtmanfred — APPROVED** [review]

**Review by isbm — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Comment by rallytime** [comment]
@rares-pop Can you add this to the [Fluorine release notes](https://github.com/saltstack/salt/blob/develop/doc/topics/releases/fluorine.rst)?

**Comment by rares-pop** [comment]
I did add a release note, but please review it and tell me if it looks right.

File: doc/topics/releases/fluorine.rst [hunk 1]
```diff
@@ -624,3 +624,14 @@ New style for including from a parent directory:
 .. code-block:: jinja
 
     {% from '../foo' import bar %}
+
+salt-api
+============================
```
Comment (rallytime) [inline doc/topics/releases/fluorine.rst]: This is picky, but can you make this line the same length as the text?
```
salt-api
========
```

We just spent a bunch of time fixing up RST syntax errors, so we want to keep these clean :D

File: doc/topics/releases/fluorine.rst [hunk 2]
```diff
@@ -624,3 +624,14 @@ New style for including from a parent directory:
...
+
+salt-api Windows support
+--------------------------------
```
Comment (rallytime) [inline doc/topics/releases/fluorine.rst]: Same comment here.

File: doc/topics/releases/fluorine.rst [hunk 3]
```diff
@@ -624,3 +624,14 @@ New style for including from a parent directory:
...
+
+Previously, salt-api was was not supported on the Microsoft Windows platforms. Now it is!
+salt-api is providing a RESTful interface to a running Salt system. It allows
```
Comment (rallytime) [inline doc/topics/releases/fluorine.rst]: `salt-api is providing` should be `salt-api provides`

File: doc/topics/releases/fluorine.rst [hunk 4]
```diff
@@ -624,3 +624,14 @@ New style for including from a parent directory:
...
+for viewing minions, runners, and jobs as well as running execution modules
+and runners of a running Salt system through a REST API that returns JSON.
```
Comment (rallytime) [inline doc/topics/releases/fluorine.rst]: Adding a link to the Salt-API documentation might be a good idea here. :)

**Review by rallytime — CHANGES_REQUESTED** [review]
Thanks @rares-pop!

I have a couple of very small requests. Once those are fixed up, we can get this in. :)

File: doc/topics/releases/fluorine.rst (hunk 4 above)
Comment (rares-pop) [reply doc/topics/releases/fluorine.rst]: Regarding Salt-API documentation, do you mean:
https://salt.readthedocs.io/projects/salt-api
or
https://docs.saltstack.com/topics/netapi/index.html
or something else?

Comment (rallytime) [reply doc/topics/releases/fluorine.rst]: Ah, sorry I was unclear. The second one: https://docs.saltstack.com/en/latest/topics/netapi/index.html

**Review by rallytime — APPROVED** [review]

**Pull request merged**